zerver/views/auth.py

```diff
@@ -1011,6 +1011,7 @@ def config_error_view(request: HttpRequest, error_category_name: str) -> HttpRes
         'google': {'social_backend_name': 'google', 'has_markdown_file': True},
         'github': {'social_backend_name': 'github', 'has_markdown_file': True},
         'gitlab': {'social_backend_name': 'gitlab', 'has_markdown_file': True},
+        'cognito': {'social_backend_name': 'cognito', 'has_markdown_file': True},
         'ldap': {'error_name': 'ldap_error_realm_is_none'},
         'dev': {'error_name': 'dev_not_supported_error'},
         'saml': {'social_backend_name': 'saml'},
```

Comment:
This is out of scope of this PR, but I think we need to do some refactor of this `auth.py` logic changed in this file (maybe just move it into the backend classes somehow?) - because needing to remember these somewhat obscure places to add an entry in makes the process of adding new backend more complicated and easy to mess up than necessary.

Reply:
I think we could move to backend classes and will be easier when we refactor this code
```python
    if backend in ["github", "google", "gitlab", "cognito"]:
        key_setting = "SOCIAL_AUTH_" + backend.upper() + "_KEY"
        secret_setting = "SOCIAL_AUTH_" + backend.upper() + "_SECRET"
        if not (getattr(settings, key_setting) and getattr(settings, secret_setting)):
```
to move to be a common function in `SocialAuthMixin` as we discussed. I'm waiting for #16248 to be merged as it'd conflict with the changes we'd make with this refactoring.